test/unit/ngsiv2/ngsiService/subscriptions-test.js

```diff
@@ -0,0 +1,351 @@
+/*
+ * Copyright 2016 Telefonica Investigación y Desarrollo, S.A.U
+ *
+ * This file is part of fiware-iotagent-lib
+ *
+ * fiware-iotagent-lib is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * fiware-iotagent-lib is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with fiware-iotagent-lib.
+ * If not, seehttp://www.gnu.org/licenses/.
+ *
+ * For those usages not covered by the GNU Affero General Public License
+ * please contact with::[contacto@tid.es]
+ *
+ * Modified by: Daniel Calvo - ATOS Research & Innovation
+ */
+'use strict';
+
+var iotAgentLib = require('../../../../lib/fiware-iotagent-lib'),
+    utils = require('../../../tools/utils'),
+    should = require('should'),
+    request = require('request'),
+    nock = require('nock'),
+    moment = require('moment'),
+    contextBrokerMock,
+    iotAgentConfig = {
+        logLevel: 'FATAL',
+        contextBroker: {
+            host: '192.168.1.1',
+            port: '1026',
+            ngsiVersion: 'v2'
+        },
+        server: {
+            port: 4041
+        },
+        types: {},
+        service: 'smartGondor',
+        subservice: 'gardens',
+        providerUrl: 'http://smartGondor.com',
+        deviceRegistrationDuration: 'P1M',
+        throttling: 'PT5S'
+    };
+
+describe('Subscription tests', function() {
+    beforeEach(function(done) {
+        var optionsProvision = {
+            url: 'http://localhost:' + iotAgentConfig.server.port + '/iot/devices',
+            method: 'POST',
+            json: utils.readExampleFile('./test/unit/examples/deviceProvisioningRequests/provisionMinimumDevice.json'),
+            headers: {
+                'fiware-service': 'smartGondor',
+                'fiware-servicepath': '/gardens'
+            }
+        };
+
+        nock.cleanAll();
+
+        iotAgentLib.activate(iotAgentConfig, function() {
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v1/updateContext',
```

Comment:
NGSIv2-based creation is not yet implemented?

Reply:
This mock corresponds to the device provisioning. When we discussed what features could be migrated to NGSIv2, you told me (https://github.com/telefonicaid/iotagent-node-lib/issues/527#issuecomment-337565075):

> Registration to CB on device provision time. Note that NGSIv2 registrations has not been yet implemented in CB (in fact, that piece of work is starting).

This is the reason why these mocks still use NGSIv1.

Reply:
What that comment menas is that operations such as `POST /v2/registrations`, `GET /v2/registrations/<id>` are not yet impemented in CB. However, note here you are using a entity creation operation (the entity creation done by IOTAgent at device provisioning type) which is implemented and corresponds to `POST /v2/entities`.

Is NGSIv2 entity creation implemented in the library? If yes, I'd suggest to use it insteead of  `POST /v1/updateContext`. If no, maybe this (and other usages of `POST /v1/updateContext` should be marked by a FIXME comment about "change once NGISv2 entity creation gets implemented".

Reply:
Yes, I understood that the NGSIv2 entity creation was ready in CB but that it would be better not implement the device provisioning in NGIv2 partially. 
It would be quite easy to add this but it would imply the modification in several files which are out of the scope of this PR (in my eyes). So if it is OK for you, we can mark these lines with FIXME and open a new PR to address this.

Reply:
Keep version consistence in all related operations sychronous operations (in this case, all the CB interactions triggered by the device provisioning request, i.e. registration + entity creation) is important. Good point.

Let's keep using /v1/updateContext here by the moement and go ahead with the FIXME marks to be located and changed easiliy in the future.

Reply:
Fixed in bea184b690273f33a1af77e43acbe831bf424d58